### PR TITLE
Adds ModalConfirmDelete.

### PR DESCRIPTION
**Pull request by webinista**
Closes department-of-veterans-affairs/messaging-fe#48
- Added reducers/modals.js, actions/modals.js
- Removed modal functionality from actions/compose.js and reducers/compose.js.
- Made some modal props required.
- Added ability to set an additional class name for a modal.

### CODE REVIEW

**Review by U-DON — CHANGES_REQUESTED** [review]
Mostly LGTM with some nits and questions.

File: src/js/messaging/components/Modal.jsx [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
+    super(props);
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  handleClose() {
+    this.props.onClose();
+  }
+
   render() {
+    const modalCss = classNames('rx-modal', this.props.cssClass);
```
Comment (webinista) [inline src/js/messaging/components/Modal.jsx]: Yes. The plan is to make it a common component.

File: src/js/messaging/containers/Compose.jsx [hunk 2]
```diff
@@ -43,71 +51,88 @@ class Compose extends React.Component {
...
   }
 
+  handleDeleteDraft() {
+    this.props.toggleConfirmDelete();
+  }
+
+  handleDelete(domEvent) {
+    // TODO: Dispatch an action that makes this API call
+    domEvent.preventDefault();
+    browserHistory.push(paths.BASE_URL);
```
Comment (webinista) [inline src/js/messaging/containers/Compose.jsx]: Actually, according to the prototype, it should go back to the drafts view not the Inbox. I need to fix that.

File: src/js/messaging/containers/Compose.jsx [hunk 3]
```diff
@@ -24,13 +26,19 @@ import {
...
 class Compose extends React.Component {
   constructor() {
     super();
     this.handleCategoryChange = this.handleCategoryChange.bind(this);
+    this.handleMessageChange = this.handleMessageChange.bind(this);
     this.handleRecipientChange = this.handleRecipientChange.bind(this);
     this.handleSubjectChange = this.handleSubjectChange.bind(this);
-    this.handleMessageChange = this.handleMessageChange.bind(this);
+    this.handleDeleteDraft = this.handleDeleteDraft.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
```
Comment (webinista) [inline src/js/messaging/containers/Compose.jsx]: `handleDeleteDraft` triggers the confirmation modal. `handleDelete` handles the actual deletion. Removed `handleDeleteDraft` in favor of using `toggleDeleteConfirm` in both places.

**Review by U-DON — APPROVED** [review]

**Comment by webinista** [comment]
Is this okay to merge? (If so, please do the honors.)

**Pull request merged**